operations/fetch.go

```diff
@@ -466,6 +467,20 @@ func fileNameWithIndex(filename string, index int) string {
 	return fmt.Sprintf("%s_(%d).%s", parts[0], index-1, strings.Join(parts[1:], "."))
 }
 
+// truncateFilename truncates the filename (minus any extensions) so the entire filename length is less than the max
+func truncateFilename(fileName string) string {
+	if len(fileName) > fileNameMaxLength {
+		parts := strings.Split(fileName, ".")
+		if len(parts) > 0 {
+			toTruncate := len(fileName) - fileNameMaxLength
+			newEndIdx := len(parts[0]) - toTruncate
+			parts[0] = parts[0][0:newEndIdx]
+			fileName = strings.Join(parts, ".")
+		}
+	}
+	return fileName
```

Comment:
I'd invert the logic of the if statement, so that you have the early return early/close to the if, and the "do truncate" part at the bottom in the main block?